Release v0.7.0 (#676)
grafana/xk6-browser: release notes/v0.7.0.md

xk6-browser v0.7.0 is here! :tada:

We've been busy preparing for big changes in the future, so this minor release is smaller than usual. It contains a new feature to unblock users using `dialog` boxes, and internal improvements.

## New features

- `xk6-browser` will now automatically dismiss dialog boxes (`alert`, `confirm`, `prompt`, and `beforeunload`). ([#663](https://github.com/grafana/xk6-browser/pull/663))

## Improvements

> - You can now launch a new browser without defining an empty options object. ([#649](https://github.com/grafana/xk6-browser/pull/649))

Reviewer: ```suggestion
- Improved launching of a browser without providing any options. ([#649](https://github.com/grafana/xk6-browser/pull/649))
 
  We couldn't launch a new browser without providing an empty object. We've fixed this, so now you can launch a new browser without an empty object.
```

    
    With the empty object:

    ```js
    const browser = chromium.launch({});
    ```

    Without the empty object:

    ```js
    const browser = chromium.launch();
    ```

### Documentation

> - Updated the video to match what we released in `v0.6.0`, and it's still relevant for `v0.7.0`. ([#639](https://github.com/grafana/xk6-browser/pull/639))

Reviewer: ```suggestion
- Updated the demo video on our README. ([#639](https://github.com/grafana/xk6-browser/pull/639))

  Now it shows the usage for xk6-browser version v0.6.0 and above.
```

> - Simplified the README. You will now be redirected to the [documentation](https://k6.io/docs/javascript-api/xk6-browser/). ([#656](https://github.com/grafana/xk6-browser/pull/656))

Reviewer: ```suggestion
- Simplified the README. ([#656](https://github.com/grafana/xk6-browser/pull/656))

  You can now find the documentation [here](https://k6.io/docs/javascript-api/xk6-browser/).
```

> ## Improvements
> 
> - You can now launch a new browser without defining an empty options object. ([#649](https://github.com/grafana/xk6-browser/pull/649))
>     
>     With the empty object:
> 
>     ```js
>     const browser = chromium.launch({});
>     ```
> 
>     Without the empty object:
> 
>     ```js
>     const browser = chromium.launch();
>     ```
> 
> 
> ### Documentation
> 
> - Updated the video to match what we released in `v0.6.0`, and it's still relevant for `v0.7.0`. ([#639](https://github.com/grafana/xk6-browser/pull/639))
> 
> - Simplified the README. You will now be redirected to the [documentation](https://k6.io/docs/javascript-api/xk6-browser/). ([#656](https://github.com/grafana/xk6-browser/pull/656))
> 
> - The project roadmap has been updated. ([#658](https://github.com/grafana/xk6-browser/pull/658))

Reviewer: I made some improvements as we discussed, and rearranged some of the items.

Feel free to commit this and then make your own changes :)

```suggestion
## Improvements

- Improved launching of a browser without providing any options. ([#649](https://github.com/grafana/xk6-browser/pull/649))
 
  We couldn't launch a new browser without providing an empty object. We've fixed this, so now you can launch a new browser without an empty object. 

    With the empty object:

    ```js
    const browser = chromium.launch({});
    ```

    Without the empty object:

    ```js
    const browser = chromium.launch();
    ```

### Documentation

- Updated the demo video on our README. ([#639](https://github.com/grafana/xk6-browser/pull/639))

  Now it shows the usage for xk6-browser version v0.6.0 and above.

- Simplified the README. ([#656](https://github.com/grafana/xk6-browser/pull/656))

  You can now find the documentation [here](https://k6.io/docs/javascript-api/xk6-browser/).

- The project roadmap has been updated. ([#658](https://github.com/grafana/xk6-browser/pull/658))
```

> - The project roadmap has been updated. ([#658](https://github.com/grafana/xk6-browser/pull/658))

Reviewer: ```suggestion
- Updated the project roadmap. ([#658](https://github.com/grafana/xk6-browser/pull/658))
```

## Internals

- Upgraded `k6` dependency to `v0.42.0`. ([#?](https://github.com/grafana/xk6-browser/pull/?))

> - Refactored the code that handles the lifecycle events. This shouldn't change any of the existing behavior, but if you notice anything that doesn't work as you expected please let us know. ([#647](https://github.com/grafana/xk6-browser/pull/647), [#644](https://github.com/grafana/xk6-browser/pull/644))

Reviewer: Added the explanation below and "actionized" what we did :) (_Improved ...._)

```suggestion
- Improved handling of `networkIdle` lifecycle events. ([#647](https://github.com/grafana/xk6-browser/pull/647), [#644](https://github.com/grafana/xk6-browser/pull/644))

  Before this, a test script would time out when xk6-browser navigates to a website and waits for the `networkIdle` lifecycle event. This shouldn't change any existing behavior, but if you notice anything that doesn't work as expected, please let us know. 
```

Author: This was the refactoring, the `networkidle` fix was done in the previous release. I'll work with your suggestion 👍

- Fix for CI. ([#674](https://github.com/grafana/xk6-browser/pull/674))

### Experimental Module Merge into k6

- `xk6-browser` is now importable as an experimental module from `k6`. ([#667](https://github.com/grafana/xk6-browser/pull/667))

> - Disable `xk6-browser` in restricted environments. ([#665](https://github.com/grafana/xk6-browser/pull/665))

Reviewer: ```suggestion
- Disabled `xk6-browser` in restricted environments. ([#665](https://github.com/grafana/xk6-browser/pull/665))
```

> - `k6` Hosts incompatibility fix. ([#671](https://github.com/grafana/xk6-browser/pull/671))

Reviewer: ```suggestion
- Fixed `k6` network `Hosts` incompatibility. ([#671](https://github.com/grafana/xk6-browser/pull/671))
```

 
  We couldn't launch a new browser without providing an empty object. We've fixed this, so now you can launch a new browser without an empty object.
    
    With the empty object:

    ```js
    const browser = chromium.launch({});
    ```

    Without the empty object:

    ```js
    const browser = chromium.launch();
    ```

### Documentation

- Updated the demo video on our README. ([#639](https://github.com/grafana/xk6-browser/pull/639))

  Now it shows the usage for xk6-browser version v0.6.0 and above.

- Simplified the README. ([#656](https://github.com/grafana/xk6-browser/pull/656))

  You can now find the documentation [here](https://k6.io/docs/javascript-api/xk6-browser/).

- Updated the project roadmap. ([#658](https://github.com/grafana/xk6-browser/pull/658))

## Internals

- Upgraded `k6` dependency to `v0.42.0`. ([#?](https://github.com/grafana/xk6-browser/pull/?))

- Refactored the lifecycle events code. ([#647](https://github.com/grafana/xk6-browser/pull/647), [#644](https://github.com/grafana/xk6-browser/pull/644))
	
    This shouldn't change any existing behavior, but if you notice anything that doesn't work as expected, please let us know. 

> - Fix for CI. ([#674](https://github.com/grafana/xk6-browser/pull/674))

Reviewer 2: ```suggestion
- Cleanup unused dependencies. ([#674](https://github.com/grafana/xk6-browser/pull/674))
```

> ### Experimental Module Merge into k6

Reviewer 2: I believe this is a big upcoming change that is quite important to the users. We plan to include `xk6-browser` into the standard distribution of `k6` as an experimental module, presumably starting from `k6@v0.43.0`. I'd suggest that we add a short description of what's going to happen and what will change for the users of `xk6-browser`. And maybe move this section to the top of the release notes and emphasize that this change is upcoming.

Author: Resolved in: 549ef0c821abbc93354ae422684da4131b0115f8

> - `xk6-browser` is now importable as an experimental module from `k6`. ([#667](https://github.com/grafana/xk6-browser/pull/667))

Reviewer 2: I believe this might create wrong impression that `xk6-browser` can now be imported as `"k6/experimental/browser"`, while the PR is only prevents the browser extension from being registered automatically as `k6/x/browser`.

Author: Resolved in: 6e87b3094a95af9b5270e7f4e7c42e67e3f7fd3a

> - Disabling of `xk6-browser` in restricted environments. ([#665](https://github.com/grafana/xk6-browser/pull/665))

Reviewer 2: ```suggestion
- `xk6-browser` can now be disabled via configuration, e.g. in restricted environments. ([#665](https://github.com/grafana/xk6-browser/pull/665))
```
